### DISCUSSION TITLE
Access to Postgresql in All-in-One using Airbyte

### DISCUSSION THREAD

**Discussion by collex100** [nextcloud/all-in-one#3925] [Questions] [answered]
Friends, please tell me how to access the Postgresql database in All-in-One using Arbyte. 
Applications are located on different VPS and can be accessed from the same host network. 

Then I want to send the data to Redash BI or SuperSet BI. 

Arbyte has the [following access details](https://docs.airbyte.com/integrations/sources/postgres).

![image](https://github.com/nextcloud/all-in-one/assets/11208423/3b33e1d6-7461-41df-9025-fb22b06ebaa6)

I [found out](https://github.com/nextcloud/all-in-one#phpmyadmin-adminer-or-pgadmin) from the help that access to the postgresql all-in-one database is as follows:
I need to connect 
the container to the docker network `nextcloud-aio` and 
use `nextcloud-aio-database` as database host, 
`oc_nextcloud` as database username and 
the password

Thank you for your help or advice, recommendation.

**Comment by szaimen** [comment]
> Friends, please tell me how to access the Postgresql database in All-in-One using Arbyte.

Hi, why do you want to do this?

**Reply by collex100** [reply]
Thanks for the question!

With the community that I am currently developing, NextCloud app to increase productivity and creativity in the development of materials. 

We need to visualize our work in NextCloud to understand whether we are achieving our goals or not.

We use a Documentary Portal, Notes, Circles, Deck (tasks) Calendar, Forms, Conference and other . BI is necessary to measure the performance of the team. 

Monitor the activity of working on documents and tasks. In general, we need a way to visualize the work in the NextCloud. 

This is how we can understand how things are going for both team leaders and team members. 

For any human activity, it is necessary to evaluate what is happening with the project and make plans based on this. 

Business intelligence is very necessary for the NextCloud app.

**Reply by szaimen** [reply, marked as answer]
As long as arbyte does not write to the database, the following should indeed work:

> I [found out](https://github.com/nextcloud/all-in-one#phpmyadmin-adminer-or-pgadmin) from the help that access to the postgresql all-in-one database is as follows:
> I need to connect
> the container to the docker network `nextcloud-aio` and
> use `nextcloud-aio-database` as database host,
> `oc_nextcloud` as database username and
> the password

**Reply by collex100** [reply]
Different data format for access. 
As I understand it, we need some kind of connector that can output port 5432 and an IP address so that LTE or BI can connect to the data in the database.

**Reply by collex100** [reply]
I have a question. If you have any recommendations, I will be grateful for your advice.

**I have studied the All-in-one installation and see the following:**

1. Port 5432 is open inside the installation.

![1024-216](https://github.com/nextcloud/all-in-one/assets/11208423/c2927a26-8f1a-402a-ab15-8e61dd76d90e)

This means that I can (I think so, but not the fact, here I have a question.) connect to Postgres from the internal network. 

2. The host is known. There is a database, a user and a password. 

1). use `nextcloud-aio-database `as database host

2). use `oc_nextcloud` as database username

3). use `sudo docker exec nextcloud-aio-nextcloud grep dbpassword config/config.php` as the password

3. There is also access to the postgres configuration, as shown in the volume. 
![388-546-max](https://github.com/nextcloud/all-in-one/assets/11208423/38ef36b8-74b4-434d-b512-48a72ed47665)

4. To access the database, it remains to execute the following commands in the postgres shell.

`CREATE USER <user_name> PASSWORD 'your_password_here';`

`GRANT USAGE ON SCHEMA <schema_name> TO <user_name>; `

`GRANT SELECT ON ALL TABLES IN SCHEMA <schema_name> TO <user_name>; `

`ALTER DEFAULT PRIVILEGES IN SCHEMA <schema_name> GRANT SELECT ON TABLES TO <user_name>;`

**As a result:** 

1. Internal port 5432 is open 

2. Access data is available (need to check) from the internal network.

3. The postgresql configuration is available for configuration.

4. I  need to understand how to create read_only user.

If you have an understanding of points 2 and 4, please write me. 

Thank you!
